realm/src/main/java/io/realm/internal/Row.java

```diff
@@ -112,8 +113,174 @@
 
     /**
      * Returns {@code true} if the field name exists.
+     *
      * @param fieldName Field name to check.
      * @return {@code true} if field name exists, {@code false} otherwise.
      */
     boolean hasColumn(String fieldName);
+
+    Row EMPTY_ROW = new Row() {
+        private final static String UNLOADED_ROW_MESSAGE = "Can't modify an row that hasn't been loaded, make sure the instance" +
+                " is loaded by calling RealmObject.isLoaded";
+
+        @Override
+        public long getColumnCount() {
+            return 0;
+        }
+
+        @Override
+        public String getColumnName(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public long getColumnIndex(String columnName) {
+            return 0;
+        }
+
+        @Override
+        public ColumnType getColumnType(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public Table getTable() {
+            return null;
+        }
+
+        @Override
+        public long getIndex() {
+            return 0;
+        }
+
+        @Override
+        public long getLong(long columnIndex) {
+            return 0;
+        }
+
+        @Override
+        public boolean getBoolean(long columnIndex) {
+            return false;
+        }
+
+        @Override
+        public float getFloat(long columnIndex) {
+            return 0;
+        }
+
+        @Override
+        public double getDouble(long columnIndex) {
+            return 0;
+        }
+
+        @Override
+        public Date getDate(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public String getString(long columnIndex) {
+            return "";
+        }
+
+        @Override
+        public byte[] getBinaryByteArray(long columnIndex) {
+            return new byte[0];
+        }
+
+        @Override
+        public Mixed getMixed(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public ColumnType getMixedType(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public long getLink(long columnIndex) {
+            return 0;
+        }
+
+        @Override
+        public boolean isNullLink(long columnIndex) {
+            throw new IllegalAccessError(UNLOADED_ROW_MESSAGE);
+        }
+
+        @Override
+        public boolean isNull(long columnIndex) {
+            throw new IllegalAccessError(UNLOADED_ROW_MESSAGE);
+        }
+
+        @Override
+        public void setNull(long columnIndex) {
+            throw new IllegalAccessError(UNLOADED_ROW_MESSAGE);
+        }
+
+        @Override
+        public LinkView getLinkList(long columnIndex) {
+            return null;
+        }
+
+        @Override
+        public void setLong(long columnIndex, long value) {
+            throw new IllegalAccessError(UNLOADED_ROW_MESSAGE);
```

Comment:
Isn't throwing an error kinda harsh? Should this be "IllegalStateException()" instead?